test/mail_spam_bounced.bats

```diff
@@ -44,10 +48,13 @@ function setup_file() {
 
 @test "checking amavis: spam message is bounced (rejected), undefined SPAMASSASSIN_SPAM_TO_INBOX should raise a warning" {
   # SPAMASSASSIN_SPAM_TO_INBOX=0 is the default. If no explicit ENV value is set, it should log a warning at startup.
-  local TEST_ENV_FILE="${PRIVATE_CONFIG}/undefined.env"
-  echo 'ENABLE_SPAMASSASSIN=1' > "${TEST_ENV_FILE}"
 
-  common_container_setup "${TEST_ENV_FILE}"
+  # shellcheck disable=SC2034
+  local TEST_DOCKER_ARGS=(
+    --env ENABLE_SPAMASSASSIN=1
+  )
+
+  common_container_setup TEST_DOCKER_ARGS
```

Comment:
`TEST_DOCKER_ARGS` is not expanded here, but just a string. Indented? I mean, it's a local variable, so it cannot be used by other functions.

Reply:
> I mean, it's a local variable, so it cannot be used by other functions.

`local` vars are scoped to the function bounds, they are visible to any function called by them IIRC, I can assure you this works :)

As an arg, it is passing just the string identifier, should I make that more clear by quote wrapping it? It's effectively a reference to the local array var.

Reply:
Maybe I missunderstood what you are trying to achieve here, but this might what you are looking for:

```sh
function common_container_setup() {
	docker run \
	--option1  \
	--option2  \
	"$@" # additional options
}

OPTIONS=("--env" "ENABLE_SPAMASSASSIN=1")
common_container_setup "${OPTIONS[@]}"
```

Reply:
I believe I considered that at the time, but wanted to keep flexibility as this functionality is still being iterated on as I rewrite tests to adopt it.

Using the array reference I can still manipulate the array on either end IIRC, but I probably chose the approach to make it a single arg, should I need to change what inputs it accepts again (_switched away from an env filepath arg in this PR_).

Looking at my rewritten `letsencrypt` test I don't see anything requiring that, so I could switch to expanding the array for now if preferred?

Reply:
Using a real string with `'...'` here may be a good idea to make it explicit. And a bit of documentation as well, because some questions come up on what this code does. Don't get me wrong, seems like good code to me, just difficult for me to understand because I'm not fluent in BATS.